### PR TITLE
Hotfix 1.35.1

### PR DESCRIPTION
**Pull request by mr0re1**
Hotfix includes:

* Bring https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2670 that was made required due to https://github.com/GoogleCloudPlatform/slurm-gcp/pull/158 being part of SlurmGCP `6.5.8` 
* [Remove `zones` from a3mega BP](https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2716/commits/402993c57a7a8cb1ddb843159683d8972eee9041) - since they don't work with `reservation` after https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2640
* Bump SlurmGCP version to `6.5.9` to get following fix:
  * https://github.com/GoogleCloudPlatform/slurm-gcp/pull/173
  Also brings along following changes:
  * https://github.com/GoogleCloudPlatform/slurm-gcp/pull/165
  * https://github.com/GoogleCloudPlatform/slurm-gcp/pull/164
  * https://github.com/GoogleCloudPlatform/slurm-gcp/pull/163

### CODE REVIEW

**Comment by nick-stroud** [comment]
Of failing tests:
- PR-test-slurm-gcp-v6-rocky8 - test completed, failed on destroy, consider passed
- PR-test-ml-gke - should not be effected by this change - ignore
- PR-test-spack-gromacs - collision on deploy, not indicative of a problem but not consider passed

**Review by ankitkinra — DISMISSED** [review]

File: examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml
Comment (nick-stroud) [inline examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml:1]: What is the reason for this change in this hot fix?

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/variables.tf
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/variables.tf:1]: Please add PR description.

File: examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml
Comment (mr0re1) [reply examples/machine-learning/a3-megagpu-8g/slurm-a3mega-cluster.yaml:1]: updated PR description

* [Remove `zones` from a3mega BP](https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2716/commits/402993c57a7a8cb1ddb843159683d8972eee9041) - since they don't work with `reservation` after https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/2640

**Review by nick-stroud — APPROVED** [review]

**Review by rohitramu — APPROVED** [review]
